packages/pigeon/lib/generator_tools.dart

```diff
@@ -432,3 +432,39 @@ Iterable<EnumeratedClass> getCodecClasses(Api api, Root root) sync* {
 /// Returns true if the [TypeDeclaration] represents an enum.
 bool isEnum(Root root, TypeDeclaration type) =>
     root.enums.map((Enum e) => e.name).contains(type.baseName);
+
+/// Formats documentation comments and adds them to current Indent.
+void addDocumentationComments(Indent indent, List<String> comments, String open,
+    {String? close, List<String>? additionalComments}) {
+  if (comments.isEmpty && additionalComments == null) {
```

Comment:
Note that this is a bug caused by allowing nullable collections, because you aren't returning if `additionalComments` is empty but non-null.